lms/static/js/views/fields.js

```diff
@@ -495,6 +495,213 @@
+                    }
+                });
+            },
+
+            imageChangeSucceeded: function (e, data) {
+                this.render();
+            },
+
+            imageChangeFailed: function (e, data) {
+            },
+
+            showImageChangeFailedMessage: function (status, responseText) {
+            },
+
+            fileSelected: function (e, data) {
+                if (_.isUndefined(data.files[0].size) || this.validateImageSize(data.files[0].size)) {
+                    data.formData = {file: data.files[0]};
+                    this.setCurrentStatus('uploading');
+                    this.setRemoveButtonVisibility('none');
+                    this.showUploadInProgressMessage();
+                    data.submit();
+                }
+            },
+
+            validateImageSize: function (imageBytes) {
+                var humanReadableSize;
+                if (imageBytes < this.options.imageMinBytes) {
+                    humanReadableSize = this.bytesToHumanReadable(this.options.imageMinBytes);
+                    this.showErrorMessage(interpolate_text(gettext("Your image must be at least {size} in size."), {size: humanReadableSize}));
+                    return false;
+                } else if (imageBytes > this.options.imageMaxBytes) {
+                    humanReadableSize = this.bytesToHumanReadable(this.options.imageMaxBytes);
+                    this.showErrorMessage(interpolate_text(gettext("Your image must be smaller than {size} in size."), {size: humanReadableSize}));
+                    return false;
+                }
+                return true;
+            },
+
+            showUploadInProgressMessage: function () {
+                this.$('.u-field-upload-button').css('opacity', 1);
+                this.$('.upload-button-icon').html(this.iconProgress);
+                this.$('.upload-button-title').html(this.titleUploading);
+            },
+
+            showRemovalInProgressMessage: function () {
+                this.$('.u-field-remove-button').css('opacity', 1);
+                this.$('.remove-button-icon').html(this.iconProgress);
+                this.$('.remove-button-title').html(this.titleRemoving);
+            },
+
+            setCurrentStatus: function (status) {
+                this.$('.image-wrapper').attr('data-status', status);
+            },
+
+            getCurrentStatus: function () {
+                return this.$('.image-wrapper').attr('data-status');
+            },
+
+            inProgress: function() {
+                var status = this.getCurrentStatus();
+                return _.isUndefined(status) ? false : true;
+            },
+
+            watchForPageUnload: function () {
+                $(window).on('beforeunload', this.onBeforeUnload);
+            },
+
+            onBeforeUnload: function () {
+                console.log('Do you really want to go away?');
```

Comment:
It is only visible in the developer console window, but I would suggest removing it anyway.